### PR TITLE
Add select to FusionTranslation.

### PR DESCRIPTION
**Pull request by rdspring1**
This PR adds support for `select` operation to FusionTranslation.

Motivation:
- If the `index` argument to `index_select` is a scalar TensorView, a `SelectOp` expression is created in the Fusion.
- Rather than map `SelectOp` to an `IndexSelectOpRecord`, this PR creates the `SelectOpRecord` for a one-to-one mapping.
- It also adds the `select` operation to python_frontend. It is functionally equivalent to [torch.select](https://pytorch.org/docs/stable/generated/torch.select.html).

### CODE REVIEW

**Comment by rdspring1** [comment]
!build

**Comment by rdspring1** [comment]
!build

**Review by jacobhinkle — COMMENTED** [review]
LGTM but I'll let @jjsjann123 give approval since this is modifying our python interface.

**Comment by rdspring1** [comment]
!build

File: csrc/python_frontend/fusion_record.h [hunk 1]
```diff
@@ -1820,6 +1820,58 @@ struct IndexSelectOpRecord : RecordFunctor {
   int64_t dim_;
 };
 
+struct SelectOpRecord : RecordFunctor {
```
Comment (jjsjann123) [inline csrc/python_frontend/fusion_record.h:1823]: nitpick. I wish we were a bit smarter and would have `IndexSelectopRecord` be general enough to be reused here.
But we don't have to worry about that for now. Can you please add a `TODO` here for merging indexing operations?

**Review by jjsjann123 — APPROVED** [review]
:clap:

**Comment by rdspring1** [comment]
!build

**Pull request merged**